addons/mail/static/src/core/common/message.js

```diff
@@ -100,6 +101,7 @@ export class Message extends Component {
         super.setup();
         this.escape = escape;
         this.popover = usePopover(this.constructor.components.Popover, { position: "top" });
+        this.props.messageEdition ??= useMessageEdition();
```

Comment:
Note: in master, everything should go through `messageEdition` instead of having multiple states corresponding to the edited message (local message component's state and global hook state). This fallback ensures that every usage of message, even those not using `messageEdition` props will keep working as before.